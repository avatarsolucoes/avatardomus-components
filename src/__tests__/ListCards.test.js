import React from 'react';
import { shallow } from '../setupEnzyme';
import { ListCards, CardItem, CardTitle, CardContent } from '..';

function TestAll() {
  return (
    <ListCards>
      <CardItem w={50}>
        <CardTitle>
          <h3>Test titulo</h3>
        </CardTitle>
        <CardContent>A</CardContent>
      </CardItem>
      <CardItem w={50} desk={50} mobile={100}>
        <CardContent>B</CardContent>
      </CardItem>
      <CardItem mobile={100}>
        <CardContent>C</CardContent>
      </CardItem>
      <CardItem>
        <CardContent>D</CardContent>
      </CardItem>
    </ListCards>
  );
}

describe('ListCards', () => {
  it('is truthy', () => {
    expect(ListCards).toBeTruthy();
  });

  it('There should be elements', () => {
    const wrapper = shallow(<TestAll />);

    // expect(wrapper.find('.card-item')).toBe(true);
    // expect(wrapper.exists('.card-offset')).toBe(true);
    // expect(wrapper.exists('.card-item-content')).toBe(true);
    // expect(wrapper.exists('.list-of-cards')).toBe(true);
    // expect(wrapper.exists('.card-item-title')).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Test renderer whith children', () => {
    const wrapper = shallow(<TestAll />).html();
    expect(wrapper).toMatchSnapshot();
  });
});
