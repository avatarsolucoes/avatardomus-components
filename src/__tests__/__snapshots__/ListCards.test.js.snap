// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCards Test renderer whith children 1`] = `"<div class=\\"list-of-cards\\"><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-title\\"><h3>Test titulo</h3></div><div class=\\"card-item-content\\" style=\\"width:auto\\">A</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">B</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">C</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">D</div></div></div></div>"`;

exports[`ListCards There should be elements 1`] = `"<div class=\\"list-of-cards\\"><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-title\\"><h3>Test titulo</h3></div><div class=\\"card-item-content\\" style=\\"width:auto\\">A</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">B</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">C</div></div></div><div class=\\"card-offset\\"><div class=\\"card-item\\" style=\\"width:auto\\"><div class=\\"card-item-content\\" style=\\"width:auto\\">D</div></div></div></div>"`;
